src/loaders/gtdb/compute_genome_attribs.py

```diff
@@ -0,0 +1,369 @@
+        f'{len(genome_ids)} genomes')
+
+
+def _run_gtdb_tk_classify_wf(batch_file_path, work_dir, debug, genome_ids, program_threads):
+    # run GTDB-TK classify_wf workflow
+    start = time.time()
+
+    # RUN gtdbtk classify_wf
+    command = ['gtdbtk', 'classify_wf',
+               '--batchfile', batch_file_path,
+               '--out_dir', work_dir,
+               '--force',
+               '--cpus', str(program_threads)]
+    command.append('--debug') if debug else None
+    print(f'running {" ".join(command)}')
+    _run_command(command, debug=debug, log_dir=os.path.join(work_dir, 'classify_wf_log'))
+    end_time = time.time()
+    print(
+        f'Used {round((end_time - start) / 60, 2)} minutes to execute gtdbtk classify_wf for '
+        f'{len(genome_ids)} genomes')
+
+
+def gtdb_tk(genome_ids, work_dir, source_data_dir, debug, program_threads, batch_number,
+            node_id, run_steps=False):
+    # NOTE: Require defining the 'GTDBTK_DATA_PATH' environment variable
+    #       e.g. export GTDBTK_DATA_PATH=/global/cfs/cdirs/kbase/collections/libraries/gtdb_tk/release207_v2
+    #       https://ecogenomics.github.io/GTDBTk/installing/index.html#gtdb-tk-reference-data
+    #
+    #       Ensure that Third-party software are on the system path.
+    #       https://ecogenomics.github.io/GTDBTk/installing/index.html#installing-third-party-software
+
+    failed_ids, size = list(), len(genome_ids)
+    print(f'Start executing GTDB-TK for {size} genomes')
+
+    batch_dir = _create_batch_dir(work_dir, batch_number, size, node_id)
+
+    # create the batch file
+    # tab separated in 2 columns (FASTA file, genome ID)
+    batch_file_name = f'genome.fasta.list'
+    batch_file_path = os.path.join(batch_dir, batch_file_name)
+
+    with open(batch_file_path, "w") as batch_file:
+        for genome_id in genome_ids:
+            genome_files = _find_genome_file(genome_id, 'genomic.fna.gz', source_data_dir)
+            # Only need xx.genomic.fna.gz file.
+            # Excluding potential xx.cds_from_genomic.fna.gz and xx.rna_from_genomic.fna.gz files
+            genome_file = [f for f in genome_files if
+                           all(name_part not in f for name_part in ['cds_from', 'rna_from', 'ERR'])]
+
+            if genome_file and len(genome_file) == 1:
+                batch_file.write(f'{genome_file[0]}\t{os.path.basename(genome_file[0])}\n')
+            else:
+                print(f'Cannot retrieve target file. Please check download folder for genome: {genome_id}')
+
+    if run_steps:
+        _run_gtdb_tk_steps(batch_file_path, batch_dir, debug, genome_ids, program_threads)
+    else:
+        _run_gtdb_tk_classify_wf(batch_file_path, batch_dir, debug, genome_ids, program_threads)
+
+    # TODO: inspect stdout for failed ids or do it in the parser program
+    return failed_ids
+
+
+def checkm2(genome_ids, work_dir, source_data_dir, debug, program_threads, batch_number, node_id):
+    # NOTE: require Python <= 3.9
```

Comment:
This seems like an issue given the pipfile is 3.11. 

Can we run via docker or something?

Reply:
I guess so. I think we have to. I haven't figured out exactly what to do.

Reply:
Ok, can you put in a TODO for now and I guess we can address in a further PR

Reply:
👍

Reply:
:+1: